src/java.base/share/classes/jdk/internal/util/OperatingSystem.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.util;
+
+import jdk.internal.util.OperatingSystemProps;
+import jdk.internal.vm.annotation.ForceInline;
+
+/**
+ * Enumeration of operating system types and testing for the current OS.
+ * The enumeration can be used to dispatch to OS specific code or values.
+ * Checking if a specific operating system is current uses a simple
+ * static method for each operating system.
+ * <p>
+ * For example,
+ * {@snippet lang = "java":
+ * if (OperatingSystem.isWindows()) {
+ *     // Windows only code.
+ * } else if (OperatingSystem.isLinux()) {
+ *     // Linux only code
+ * }
+ *}
+ *
+ * Alternatively, compare with the {@linkplain #current() current} operating system.
+ * For example,
+ * {@snippet lang = "java":
+ * if (OperatingSystem.current() == OperatingSystem.WINDOWS) {
+ *     // Windows only code.
+ * }
+ *}
+ * Dispatch based on the current operating system or choose a value.
+ * For example,
+ * {@snippet lang = "java":
+ * int port() {
+ *      return switch(OperatingSystem.current()) {
+ *          case LINUX->32768;
+ *          case AIX->32768;
+ *          case MACOS->49152;
+ *          case WINDOWS->49152;
+ *      };
+ * }
+ *}
+ */
+public enum OperatingSystem {
+
+    /**
+     * The Linux Operating system.
```

Comment:
For the sake if completeness, Linux isn't an operating system, so this statement isn't correct.

Reply:
A bit of a quibble, several internet sources identify Linux as an operating system (including Redhat).

Reply:
Well, no. It is just a kernel. Without a userland not an OS. That is why they call it GNU/Linux. Kernel + userland.

Reply:
If you are referring to "Red Hat Enterprise Linux", it'd be correct to say that Red Hat identifies "Red Hat Enterprise Linux" as an operating system, but I wouldn't go so far as to say that Red Hat calls "Linux" an operating system (well... subject to the reality of human inaccuracy). And the validity and correctness of the term "GNU/Linux" is *definitely* disputed (and likely always will be).

Maybe "Operating systems based on the Linux kernel" would be satisfactory?

Reply:
Not the rabbit-hole I expected to go down today or for an informal comment on an internal API.
I have no skin in that game.
https://www.redhat.com/en/topics/linux/what-is-linux

Reply:
Joke, they should know it better especially because they use the registered trademark symbol. Really surprising.